keras/utils/generic_utils.py

```diff
@@ -9,10 +9,21 @@
 import marshal
 import types as python_types
 
+global_custom_objects = {}
```

Comment:
This is unsafe, because it would be reused across different unrelated calls to `layers_from_config`.

I would suggest using a custom object scope inside `layers_from_config`:

```python
with custom_objects_scope(custom_objects):
   layer = ...
```

Or something like that.